### DISCUSSION TITLE
Delete Edge On Drop example only for a specific EdgeType.

### DISCUSSION THREAD

**Discussion by surya0180** [xyflow/xyflow#3244] [Q&A] [answered]
Hi team, I have gone through the Delete Edge on Drop example in the documentation and want to use that in my application. But I dont want to implement that functionality for all the edges. There some EdgeTypes defined and I want only those edge types to have that functionality. Any idea how I can achieve it? Any help would be appreciated

EDIT: Let me rephrase my question. When the onEdgeUpdateStart, OnEdgeUpdateEnd functions are given to the component, all the edges can be dragged to anywhere and can be dropped to some other node or can be dropped in the panel so that the edge will be deleted.

What I want is to have this drag functionality only for some edge types. I don't want all my edges to be draggable

**Comment by hayleigh-dot-dev** [comment]
Hi there, yeah this should be quite simple to implement! The quickest way to get something working by modifying the example would be to add an extra check to `onEdgeUpdateEnd`: you have access to the edge in the event so you can perform any logic you want there.

```js
const onEdgeUpdateEnd = useCallback((_, edge) => {
  if (!edgeUpdateSuccessful.current && edge.type !== '...') {
    setEdges((eds) => eds.filter((e) => e.id !== edge.id));
  }

  edgeUpdateSuccessful.current = true;
}, []);
```

Let me know if this doesn't work!

**Reply by surya0180** [reply]
Hi @hayleigh-dot-dev . Thanks for the reply but I was expecting something different. Let me modify my question and type it here as well. 
When the onEdgeUpdateStart, OnEdgeUpdateEnd functions are given to the <Reactflow /> component, all the edges can be dragged to anywhere and can be dropped to some other node or can be dropped in the panel so that the edge will be deleted. 

What I want is to have this drag functionality only for some edge types. I don't want all my edges to be draggable. I hope I am being clear with my doubt 😓 .

**Reply by bcakmakoglu** [reply, marked as answer]
Are you possibly looking for the `edge.updatable` option? 

```ts
  { id: 'e1-2', source: '1', target: '2', updatable: false },
```

**Reply by surya0180** [reply]
Hi @bcakmakoglu ! I remember you answered my previous discussion as well. Thanks a lot for the answer. Being completely new to the library it's hard to find a fix some times. Lucky that this community is completely active.

**Reply by surya0180** [reply]
Also thanks to you @hayleigh-dot-dev , I can simply add that condition instead of creating a custom edge just for making an edge non draggable 😃

**Reply by surya0180** [reply]
I'm closing this discussion now since it is resolved.
